modules/internal/ChapelArray.chpl

```diff
@@ -3211,6 +3211,68 @@ module ChapelArray {
       else
         a.add(e);
   }
+
+  //
+  // BaseSparseDom operator overloads
+  //
+  proc +=(ref sd: domain, inds: [] sd.idxType) where isSparseDom(sd) &&
+    sd.rank==1 {
+
+    if inds.size == 0 then return;
+
+    sd._value.dsiBulkAdd(inds);
+  }
+
+  proc +=(ref sd: domain, inds: [] sd.rank*sd.idxType) where isSparseDom(sd) &&
+    sd.rank>1 {
+
+    if inds.size == 0 then return;
+
+    sd._value.dsiBulkAdd(inds);
+  }
+
+  // Currently this is not optimized for addition of a sparse
+  // This assumes that a domain is sorted by row, which may not always be true.
+  // There could be domains sorted by column instead.
+  // It would be nice if there was a common interface in BaseDom to query
+  // whether a domain was row-major or column-major to distinguish these.
+  // The current implementation is a work-around for this shortcoming.
+  proc +=(ref sd: domain, d: domain)
+  where sd._value: DefaultSparseDom && d.rank==sd.rank && sd.idxType==d.idxType {
+
+    if d.size == 0 then return;
+
+    type _idxType = if sd.rank==1 then int else sd.rank*int;
+    const indCount = d.numIndices;
+    const arr: [{0..#indCount}] _idxType;
```

Comment:
This is another case where I think we could use `index(sd)` for simplicity in place of computing `_idxType`.  I feel surprised that `_idxType` is defined in terms of `int` rather than the `idxType` of `d` or `sd`.